devito/propagator.py

```diff
@@ -200,7 +201,7 @@ def sympy_to_cgen(self, stencils):
                 # TODO: add support for double precision
                 self.add_local_var(name, np.float32)
                 # TODO: undo precision enforcing
-                factors.append(cgen.Assign(name, str(ccode(expr.xreplace(self._var_map))).replace("pow", "powf").replace("fabs", "fabsf")))
+                factors.append(cgen.Assign(name, str(ccode(self.time_substitutions(expr).xreplace(self._var_map))).replace("pow", "powf").replace("fabs", "fabsf")))
```

Comment:
The method `time_substitutions` replaces `t+s` with `t1`, for example, but `t1` is only defined when `save=False`. `t1` doesn't and shouldn't exist when `save=True`. `t+s` just translates to `t+1` in case of `save=True`, a transformation that happens elsewhere. 
`time_substitutions` does not care about `_var_map`.